cmd/ctr/pprof.go

```diff
@@ -0,0 +1,181 @@
+	Name:  "goroutines",
+	Usage: "dump goroutine stack dump",
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "full,f",
+			Usage: "full goroutine stack dump",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		var uri string
+		client := getPProfClient(context)
+
+		full := context.Bool("full")
+		if full {
+			uri = "/debug/pprof/goroutine?debug=2"
+		} else {
+			uri = "/debug/pprof/goroutine?debug=1"
+		}
+		output, err := makeTCPRequest(client, uri)
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		fmt.Println(output)
+		return err
+	},
+}
+
+var pprofHeapCommand = cli.Command{
+	Name:  "heap",
+	Usage: "dump heap profile",
+	Action: func(context *cli.Context) error {
+		client := getPProfClient(context)
+
+		output, err := makeTCPRequest(client, "/debug/pprof/heap")
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		fmt.Println(output)
+		return err
+	},
+}
+
+var pprofProfileCommand = cli.Command{
+	Name:  "profile",
+	Usage: "CPU profile",
+	Action: func(context *cli.Context) error {
+		client := getPProfClient(context)
+
+		output, err := makeTCPRequest(client, "/debug/pprof/profile")
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		fmt.Println(output)
+		return err
+	},
+}
+
+var pprofTraceCommand = cli.Command{
+	Name:  "trace",
+	Usage: "collects execution trace",
+	Flags: []cli.Flag{
+		cli.IntFlag{
+			Name:  "seconds,s",
+			Usage: "profile time (seconds)",
+			Value: 5,
+		},
+	},
+	Action: func(context *cli.Context) error {
+		client := getPProfClient(context)
+
+		seconds := context.Int("seconds")
+		uri := fmt.Sprintf("/debug/pprof/trace?seconds=%d", seconds)
+		output, err := makeTCPRequest(client, uri)
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		fmt.Println(output)
+		return err
+	},
+}
+
+var pprofBlockCommand = cli.Command{
+	Name:  "block",
+	Usage: "goroutine blocking profile",
+	Action: func(context *cli.Context) error {
+		client := getPProfClient(context)
+
+		output, err := makeTCPRequest(client, "/debug/pprof/block")
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		fmt.Println(output)
+		return err
+	},
+}
+
+var pprofThreadcreateCommand = cli.Command{
+	Name:  "threadcreate",
+	Usage: "goroutine blocking profile",
+	Action: func(context *cli.Context) error {
+		client := getPProfClient(context)
+
+		output, err := makeTCPRequest(client, "/debug/pprof/threadcreate")
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+		fmt.Println(output)
+		return err
+	},
+}
+
+func (d *pprofDialer) pprofDial(proto, addr string) (conn net.Conn, err error) {
+	return net.Dial(d.proto, d.addr)
+}
+
+func getPProfClient(context *cli.Context) *http.Client {
+	addr := context.GlobalString("debug-socket")
+	dialer := pprofDialer{"unix", addr}
+
+	tr := &http.Transport{
+		Dial: dialer.pprofDial,
+	}
+	client := &http.Client{Transport: tr}
+	return client
+}
+
+func makeTCPRequest(client *http.Client, request string) (string, error) {
```

Comment:
This is an http request.

Also, let's not suck up the whole thing into memory. The results can be quite large in practice. Just return the body as a read closer and `io.Copy` it to stdout.